vm/src/stdlib/math.rs

```diff
@@ -131,8 +131,21 @@ fn math_log1p(x: IntoPyFloat) -> f64 {
 make_math_func!(math_log2, log2);
 make_math_func!(math_log10, log10);
 
-fn math_pow(x: IntoPyFloat, y: IntoPyFloat) -> f64 {
-    x.to_f64().powf(y.to_f64())
+fn math_pow(x: IntoPyFloat, y: IntoPyFloat, vm: &VirtualMachine) -> PyResult<f64> {
+    let x = x.to_f64();
+    let y = y.to_f64();
+    
+    if x < 0.0 && x.is_finite() && y.fract() != 0.0 && y.is_finite() {
+        return Err(vm.new_value_error("math domain error".to_owned()));
+    }
+
+    if x == 0.0 && y < 0.0 {
+        return Err(vm.new_value_error("math domain error".to_owned()));
+    }
+
+    let value = x.powf(y);
+
+    return Ok(value);
```

Comment:
```suggestion
    Ok(value)
```
the last `return` statement in a block can be replaced by an expression of its value.

Reply:
Thank you for suggestion : )
Should I add a commit for this suggestion? or rewrite my commit?

Reply:
that's up to you. either amending + push force or commiting suggestion would be the same result when it will be merged by squash merging.

Reply:
I got it, thanks for explaining!